### PR TITLE
Open templates from list instead of loading the URL in block templates e2e tests

### PR DESCRIPTION
**Pull request by Aljullu**
### Changes proposed in this Pull Request:

This PR creates a new util named `editorUtils.visitTemplateEditor()`. It has some benefits:

1. It encapsulates the logic of opening a template in the site editor, entering edit mode and closing the welcome modal.
2. It simulates the user going to the template list and clicking on the template with the provided name. With the old approach, we would go directly to the template screen URL. This new approach is more similar to what a real user would do and allows us to catch bugs where the template might be listed with the incorrect title or URL.

In addition to that, this PR fixes a typo: `canBeOverridenByThemes` → `canBeOverriddenByThemes`.

### How to test the changes in this Pull Request:

Note: no need to test this for the release.

1. Verify e2e tests pass.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment

Open templates from list instead of loading the URL in block templates e2e tests

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/single-product-template.block_theme_with_templates.spec.ts [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
  */
 import { test, expect } from '@woocommerce/e2e-playwright-utils';
-import { BLOCK_THEME_WITH_TEMPLATES_SLUG } from '@woocommerce/e2e-utils';
 
 const testData = {
 	permalink: '/product/belt',
 	templateName: 'Single Product Belt',
 	templatePath: 'single-product-belt',
-	templateType: 'wp_template',
+	templateType: 'wp_template' as 'wp_template' | 'wp_template_part',
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/tests/templates/single-product-template.block_theme_with_templates.spec.ts]: Considering that we are using the `wp_template | wp_template_part` in multiple places, I wonder if wouldn't be better to define a type or an Enum for it, for example:

```typescript
enum TemplateType {
  WpTemplate = "wp_template",
  WpTemplatePart = "wp_template_part",
}
```

File: plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts [hunk 2]
```diff
@@ -372,6 +372,44 @@ export class EditorUtils {
 			.waitFor();
 	}
 
+	async visitTemplateEditor(
+		templateName: string,
+		templateType: 'wp_template' | 'wp_template_part'
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts]: We could use the enum here.

File: plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts [hunk 3]
```diff
@@ -372,6 +372,44 @@ export class EditorUtils {
...
+	) {
+		if ( templateType === 'wp_template_part' ) {
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts:384]: And here

File: plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts [hunk 4]
```diff
@@ -372,6 +372,44 @@ export class EditorUtils {
...
+			await templateButton.click();
+		}
+
+		await this.enterEditMode();
+		await this.closeWelcomeGuideModal();
+		await this.waitForSiteEditorFinishLoading();
+
+		// Verify we are editing the correct template and it has the correct title.
+		const templateTypeName =
+			templateType === 'wp_template' ? 'template' : 'template part';
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts:410]: Another place that would be possible to use the enum

**Review by thealexandrelara — APPROVED** [review]
Nice catch with the typo! I left a minor suggestion in the code, but that is not a blocker so I'm approving the PR. Thank you for working on this! 🙌

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/single-product-template.block_theme_with_templates.spec.ts (hunk 1 above)
Comment (Aljullu) [reply plugins/woocommerce-blocks/tests/e2e/tests/templates/single-product-template.block_theme_with_templates.spec.ts]: Good point! I created a `TemplateType` type in d67619be6c4787f4fb04731b8797b28d40cbf751. I don't have a strong preference between an enum and a type, though. I went with a type as there are only two values and it seemed to add less complexity, but I wouldn't mind changing it.

**Pull request merged**